Fluid/Filters/MiscFilters.cs

```diff
@@ -286,7 +287,8 @@ public static FluidValue Json(FluidValue input, FilterArguments arguments, Templ
         {
             var options = new JsonSerializerOptions
             {
-                WriteIndented = arguments.At(0).ToBooleanValue()
+                WriteIndented = arguments.At(0).ToBooleanValue(),
+                PropertyNamingPolicy = new FluidJsonNamingPolicy(context)
```

Comment:
The internal cache won't take into the account the fact that a context can have a different MemberNameStrategy though. It feels like the cache should be higher, per MemberNameStrategy  instance, and then you wouldn't need a static cache in the FluidJsonNamingPolicy. Also maybe an immutable dictionary instead of a concurrent one to prevent locking which will probably never be an issue.

Or maybe also change the naming policy contract so it can accept a name and not a member info, that would make things much easier and not require a cache.